assets/js/base/hooks/shipping/use-select-shipping-rate.js

```diff
@@ -13,21 +13,30 @@ See also: https://github.com/woocommerce/woocommerce-gutenberg-products-block/tr
  * @return {Object} This hook will return an object with two properties:
  * - selectShippingRate    A function that immediately returns the selected
  * rate and dispatches an action generator.
- * - selectedShippingRates An array of selected shipping rates, maintained
+ * - selectedShippingRates An object of selected shipping rates and package id as key, maintained
  * locally by a state and updated optimistically.
  */
 export const useSelectShippingRate = ( shippingRates ) => {
-	const initiallySelectedRates = shippingRates.map(
-		// the API responds with those keys.
-		// eslint-disable-next-line camelcase
-		( p ) => p.shipping_rates.find( ( rate ) => rate.selected )?.rate_id
+	const initiallySelectedRates = Object.fromEntries(
```

Comment:
In #1941 I noticed `Object.fromEntries()` doesn't seem to work with IE11. We should either refactor this so it doesn't use `Object.fromEntries()` or find a way for Bable to transpile this.